cli/run_util.go

```diff
@@ -580,6 +580,57 @@ func activateStepLibStep(stepIDData models.StepIDData, destination, stepYMLCopyP
 	return info, didStepLibUpdate, nil
 }
 
+func expandStepInputs(
+	inputs []envmanModels.EnvironmentItemModel,
+	environments []envmanModels.EnvironmentItemModel,
+) map[string]string {
+	stepInputs := make(map[string]string)
+
+	// Retrieve all non-sensitive input values
+	for _, environmentItem := range inputs {
+		if err := environmentItem.FillMissingDefaults(); err != nil {
+			log.Warnf("Failed to fill missing defaults, skipping input")
+			continue
+		}
+
+		options, err := environmentItem.GetOptions()
+		if err == nil && *options.IsSensitive == false {
+			if inputName, inputValue, err := environmentItem.GetKeyValuePair(); err == nil {
+				inputString := fmt.Sprintf("%v", inputValue)
+				stepInputs[inputName] = inputString
+			} else {
+				log.Warnf("Failed to get input value, skipping input")
+			}
+		} else if err != nil {
+			log.Warnf("Failed to get input options, skipping input")
+		}
+	}
+
+	var mappingFunc func(string) string
+	mappingFunc = func(key string) string {
+		for inputName, inputValue := range stepInputs {
+			if inputName == key {
+				return os.Expand(fmt.Sprintf("%v", inputValue), mappingFunc)
+			}
+		}
+
+		for _, environmentItem := range environments {
+			envValue := environmentItem[key]
+			if envValue != nil {
+				return os.Expand(fmt.Sprintf("%v", envValue), mappingFunc)
```

Comment:
NOTE: as an input's key can not be `opts` it is sure the value of `environmentItem[key]` is a string, it is might better expressed this way: `envValue .(string)`

Reply:
causes compilation error

Reply:
ok